File: 1818-maximum-score-from-removing-substrings/maximum-score-from-removing-substrings.cs
public class Solution {
    public string ReverseString(string s){
        char[] ara=s.ToCharArray();
        Array.Reverse(ara);
        return new string(ara);
    }
    public int MaximumGain(string s, int x, int y) {
        int n=s.Length;
        if(x<y){
            int t=x;
            x=y;
            y=t;
            s=ReverseString(s);
        }
        int a=0,b=0,ans=0;
        for(int i=0;i<n;i++){
            if(s[i]=='a'){
                a++;
            }
            else if(s[i]=='b'){
                if(a>=1){
                    ans+=x;
                    a--;
                }
                else{
                    b++;
                }
            }
            else{
                ans+=Math.Min(a,b)*y;
                a=0;
                b=0;
            }
        }
        ans+=Math.Min(a,b)*y;
        return ans;
    }
}

File: 1818-maximum-score-from-removing-substrings/maximum-score-from-removing-substrings.cpp
class Solution {
public:
    int maximumGain(string s, int x, int y) {
        int n=s.length();
        if(x<y){
            swap(x,y);
            reverse(s.begin(),s.end());
        }
        int ans=0,a=0,b=0;
        for(int i=0;i<n;i++){
            if(s[i]=='a'){
                a++;
            }
            else if(s[i]=='b'){
                if(a>0){
                    a--;
                    ans+=x;
                }
                else{
                    b++;
                }
            }
            else{
                ans+=min(a,b)*y;
                a=0;
                b=0;
            }
        }
        ans+=min(a,b)*y;
        return ans;
    }
};